### PR TITLE
Fix/improve search docs

### PR DESCRIPTION
**Pull request by vyasr**
## Description
Improve documentation of `find_jobs` using job documents, and add deprecation notes for `find_job_documents` and `find_statepoints`.

## Motivation and Context
Closes issue #95.

## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

This pull request is based on
- [ ] *master*, because it is a bug fix or an update to the documentation.
- [x] *develop*, because it introduces a new feature.

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.txt).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt).

### CODE REVIEW

**Comment by vyasr** [comment]
@csadorf let me know if you intended for there to be a more substantial change to the `find_jobs` documentation.

File: signac/contrib/project.py [hunk 1]
```diff
@@ -520,8 +520,10 @@ def find_job_ids(self, filter=None, doc_filter=None, index=None):
     def find_jobs(self, filter=None, doc_filter=None, index=None):
```
Comment (csadorf) [inline signac/contrib/project.py]: We will need to remember to remove the `index` argument here.

Comment (vyasr) [reply signac/contrib/project.py]: For what milestone? We can make an issue now and tag it appropriately.

Comment (csadorf) [reply signac/contrib/project.py]: For 1.0.0, I just created the issue: https://github.com/glotzerlab/signac/issues/131

**Pull request merged**